EpsilonCCS/Src/TransmitSerial.c

```diff
@@ -2,7 +2,37 @@
 
 #include "stm32f4xx_hal_uart.h"
 
+/*
+* Flashing Red LED: The USART line is has errored sending this message, will re-try
+* Solid Red LED: The USART Port has timed out, check if Hermes is on
+*/
 void transmitMessage(uint8_t* payload, uint16_t bufferSize)
 {
-    HAL_UART_Transmit(&huart3, payload, bufferSize, 5000);
+    uint16_t attemptCount = 0;
+
+    while (1)
```

Comment:
This loop is not intended to be indefinite, so this is dangerous.  `attemptCount > NUMBER_OF_ATTEMPTS` should go in here.

Reply:
You mean while(attemptCount < NUMBER_OF_ATTEMPTS) right?